src/pages/steps/AdditionalInformationStep.tsx

```diff
@@ -149,26 +152,15 @@ const AdditionalInformationStep = ({
 
   const [tab, setTab] = useState('description');
 
-  const hashHandlerCallback = useCallback(() => {
-    const { hash } = window.location;
-    const newTab = hash.replace('#', '');
-    if (newTab && Object.values(Fields).some((field) => newTab === field)) {
-      setTab(newTab);
-    }
-  }, [setTab]);
+  const [_path, hash] = asPath.split('#');
 
   useEffect(() => {
-    hashHandlerCallback();
-
-    window.addEventListener('hashchange', hashHandlerCallback);
-
-    return () => {
-      window.removeEventListener('hashchange', hashHandlerCallback);
-    };
-  }, [hashHandlerCallback]);
+    if (!hash || !Object.values(Fields).some((field) => hash === field)) return;
+    setTab(hash);
+  }, [hash]);
 
   const handleSelectTab = (tab: string) => {
-    window.location.hash = tab;
+    router.push({ hash: tab });
```

Comment:
You can probably use the shallow option here, so the server side code doesn't get run again/

https://nextjs.org/docs/routing/shallow-routing

Reply:
Done